sympy/polys/polytools.py

```diff
@@ -7938,3 +7937,399 @@ def named_poly(n, f, K, name, x, polys):
     else:
         poly = Poly.new(poly, head)
     return poly if polys else poly.as_expr()
+
+
+def factor_system(eqs, *gens, **kwargs):
+    """
+    Given a system of polynomial equations, this function returns a list of sets,
+    where each set contains irreducible factors of the equations in the system.
+    The union of solutions to each set gives the complete solution to the original system.
+
+    Parameters
+    ==========
+
+    eqs : list
+        List of expressions to be factored. Each expression is assumed to be equal to zero.
+    *gens : Symbol or sequence of Symbols, optional
+        Generator(s) of the polynomial ring. If not provided, all free symbols will be used.
+    **kwargs : dict, optional
+        Optional keyword arguments passed to polynomial manipulation functions.
+
+    Returns
+    =======
+
+    list of sets
+        Each set in the list contains expressions that, when all set to zero,
+        form one component of the solution. The complete solution is the union
+        of solutions to each set.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.polytools import factor_system
+    >>> from sympy.abc import x, y
+    >>> result = factor_system([x**2 - 1])
+    >>> sorted_result = sorted([sorted(s, key=str) for s in result], key=lambda x: [str(i) for i in x])
+    >>> sorted_result
+    [[x + 1], [x - 1]]
```

Comment:
It is not good to have this complicated `sorted` expression in the docs. Better just to show the output of `factor_system`.